src/org/usfirst/frc/team1922/robot/commands/TurnTo_Command.java

```diff
@@ -25,20 +27,40 @@ public TurnTo_Command(int direction) {
     @Override
     protected void initialize() {
         start = Robot.m_driveTrain.getAngle();
-        Robot.m_driveTrain.drive(direction*(.2), -direction*(.2));
+        if( direction >= 0 ){
+        	Robot.m_driveTrain.drive(.4,-.4);
+        }else{ 
+        	isLeft = true;
+        	Robot.m_driveTrain.drive(-.35,.35);
+        	
+        }
+        SmartDashboard.putString("Turn Direction", "" + direction);
+        SmartDashboard.putString("Start Start", "" + start);
+        
         SmartDashboard.putString("Turn_Command", "intialized");
     }
     
     @Override
     protected void execute() {
```

Comment:
We can reduce the size and complexity of this method if we think a little bit about what it does. It will also unify the way the smart dashboard output looks which will make it easier to debug.

At this point the robot is already turning, all we need to do is decide whether to keep going or not. How we determine that is different depending on whether we're turning left or turning right. The rest of the code is identical though so we don't need such a large if block.
```
boolean keepGoing = false;
if(isLeft) // again the positive if condition should be first
{
   keepGoing = Robot.m_driveTrain.getAngle() >= start + direction;
}
else
{
   keepGoing = Robot.m_driveTrain.getAngle() <= start + direction;
}

isFinished = !keepGoing;
SmartDashboard.putString("TurnTo_Command", "Turning " + keepGoing);
SmartDashboard.putString("TurnTo Current Angle", "" + Robot.m_driveTrain.getAngle());
```

Reply:
Note you could also assign directly to `isFinished` inside of the if/else by reversing the inequalities. That would be totally fine but I thought it might be harder to understand what I was getting at if I did that.